mqtt/mqtt-edgehub/tests/command.rs

```diff
@@ -0,0 +1,81 @@
+use futures_util::StreamExt;
+use mqtt3::{proto::ClientId, ShutdownError};
+use mqtt_broker::{auth::AllowAll, BrokerBuilder, BrokerHandle};
+use mqtt_broker_tests_util::{
+    client::TestClientBuilder,
+    packet_stream::PacketStream,
+    server::{start_server, DummyAuthenticator},
+};
+use mqtt_edgehub::command::{
+    CommandHandler, CommandHandlerError, ShutdownHandle as CommandShutdownHandle,
+};
+use tokio::task::JoinHandle;
+
+const TEST_SERVER_ADDRESS: &str = "localhost:5555";
+
+/// Scenario:
+// create broker
+// create command handler
+// connect client
+// publish message to disconnect client
+// verify client disconnected
+#[tokio::test]
+async fn disconnect_client() {
+    let broker = BrokerBuilder::default().with_authorizer(AllowAll).build();
+    let broker_handle = broker.handle();
+
+    let server_handle = start_server(broker, DummyAuthenticator::anonymous());
+
+    let (mut command_handler_shutdown_handle, join_handle) =
+        start_command_handler(broker_handle, TEST_SERVER_ADDRESS.to_string())
+            .await
+            .expect("could not start command handler");
+
+    let mut test_client = PacketStream::connect(
+        ClientId::IdWithCleanSession("test-client".into()),
+        server_handle.address(),
+        None,
+        None,
+    )
+    .await;
+    test_client.next().await; // skip connack
+
+    let mut edgehub_client = TestClientBuilder::new(server_handle.address())
+        .with_client_id(ClientId::IdWithCleanSession("$edgehub".into()))
+        .build();
+
+    let topic = "$edgehub/test-client/disconnect";
+    edgehub_client.publish_qos1(topic, "qos 1", false).await;
```

Comment:
hm, this test made me think... why do we put client id into the topic and use regex to extract it? can't we just put client id in the payload of the publication to "$edgehub/disconnect"?

Reply:
Vadim and I talked offline and decided that this was a good change, because then we don't need regex logic / tests. It was originally done this way for consistency. I think it is best to merge the working version now and fix later.